context/src/main/java/io/grpc/Deadline.java

```diff
@@ -164,13 +164,19 @@ public int compareTo(Deadline that) {
     return 0;
   }
 
-  /** Time source representing nanoseconds since fixed but arbitrary point in time. */
-  abstract static class Ticker {
+  /**
+   * Time source representing nanoseconds since fixed but arbitrary point in time.
+   *
+   * <p>This is public for testing purposes.
+   */
+  public interface DeadlineTicker {
     /** Returns the number of nanoseconds since this source's epoch. */
-    public abstract long read();
+    long read();
   }
 
-  private static class SystemTicker extends Ticker {
```

Comment:
We already had a singleton: `SYSTEM_TICKER`. This is a different way of doing a singleton that has the only benefit of avoiding creating the singleton if it goes unused. Since test code is the only place it would be unused, it seemed like you made the change to benefit test code.

Either that, or you just felt like changing the code for the heck of it.